File: REGROUPER/FCGP_Regrouper.vb
Imports System.Reflection
''' <summary> une seule sub en remplacement de l'initialisation de l'application par VB.
''' Correspondance directe avec C# </summary>
Friend Module FCGP_Regrouper
    ''' <summary> Point d'entrée principal de l'application. </summary>
    <STAThread>
    Friend Sub Main()
        Dim GuidID = "FCGP_REGROUPER_VB_CORE"
        'retrouve le Guid associé à l'assembly pour pouvoir rendre l'instance unique
        Dim AttributsAssembly = GetType(FCGP_Regrouper).Assembly.CustomAttributes
        For Each I As CustomAttributeData In AttributsAssembly
            If I.AttributeType.Name = "GuidAttribute" Then
                GuidID = CStr(I.ConstructorArguments(0).Value)
                Exit For
            End If
        Next
        Application.SetHighDpiMode(HighDpiMode.DpiUnaware)
        Application.SetDefaultFont(New Font("Segoe UI", 14.0!, FontStyle.Regular, GraphicsUnit.Pixel))
        'émulation de la propriété visualbasic Application.SingleInstance=true
        Using mutex As New Mutex(False, GuidID)
            If mutex.WaitOne(0) Then
                Application.EnableVisualStyles()
                Application.SetCompatibleTextRenderingDefault(False)
                Application.Run(New Regrouper)
            End If
        End Using
    End Sub
End Module

File: REGROUPER/FCGP_Regrouper.cs
using System.Reflection;

namespace FCGP
{
    /// <summary> une seule sub en remplacement de l'initialisation de l'application par VB.
    /// Correspondance directe avec C# </summary>
    internal static class FCGP_Regrouper
    {
        /// <summary> Point d'entrée principal de l'application. </summary>
        [STAThread]
        internal static void Main()
        {
            string GuidID = "FCGP_REGROUPER_CS_CORE";
            // retrouve le Guid associé à l'assembly pour pouvoir rendre l'instance unique
            var AttributsAssembly = typeof(FCGP_Regrouper).Assembly.CustomAttributes;
            foreach (CustomAttributeData I in AttributsAssembly)
            {
                if (I.AttributeType.Name == "GuidAttribute")
                {
                    GuidID = Convert.ToString(I.ConstructorArguments[0].Value);
                    break;
                }
            }
            Application.SetHighDpiMode(HighDpiMode.DpiUnaware);
            Application.SetDefaultFont(new Font("Segoe UI", 14.0f, FontStyle.Regular, GraphicsUnit.Pixel));
            // émulation de la propriété visualbasic Application.SingleInstance=true
            using (var mutex = new Mutex(false, GuidID))
            {
                if (mutex.WaitOne(0))
                {
                    Application.EnableVisualStyles();
                    Application.SetCompatibleTextRenderingDefault(false);
                    Application.Run(new Regrouper());
                }
            }
        }
    }
}